### PR TITLE
Fix incompatibility with arm architecture in olap #2645

### PR DESCRIPTION
**Pull request by yangzhg**
1. Add macros in olap code to be compatible with arm instruction set
2. Replace atomic functions  in olap whit gutils/atomicops

#2645

### CODE REVIEW

File: be/src/olap/storage_engine.h [hunk 1]
```diff
@@ -310,6 +308,8 @@ class StorageEngine {
     uint32_t _min_percentage_of_error_disk;
     Cache* _file_descriptor_lru_cache;
     Cache* _index_stream_lru_cache;
+    uint32_t _max_base_compaction_task_per_disk;
+    uint32_t _max_cumulative_compaction_task_per_disk;
```
Comment (imay) [inline be/src/olap/storage_engine.h]: uesless?

File: be/src/olap/rowset/segment_group.h [hunk 2]
```diff
@@ -291,7 +291,7 @@ class SegmentGroup {
     PUniqueId _load_id;                // load id for segmentgroup
     int32_t _num_segments;             // number of segments in this segmentgroup
     bool _index_loaded;                // whether the segmentgroup has been read
-    atomic_t _ref_count;               // reference count
+    base::subtle::Atomic64 _ref_count; // reference count
```
Comment (imay) [inline be/src/olap/rowset/segment_group.h]: prefer std::atomic<int32_t>.

File: be/src/olap/bhp_lib.h [hunk 3]
```diff
@@ -789,6 +790,11 @@ inline int check_sse4_2() {
     );
     return ret;
 }
+#else
+inline int check_sse4_2() {
```
Comment (imay) [inline be/src/olap/bhp_lib.h]: Is this can be unified through CpuInfo?

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/segment_group.h (hunk 2 above)
Comment (yangzhg) [reply be/src/olap/rowset/segment_group.h]: gutil/atomicops need this type

File: be/src/olap/storage_engine.h (hunk 1 above)
Comment (yangzhg) [reply be/src/olap/storage_engine.h]: done

**Review by imay — DISMISSED** [review]
LGTM

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**